ebms-async/src/main/kotlin/no/nav/emottak/ebms/async/processing/SignalMessageService.kt

```diff
@@ -1,75 +1,50 @@
 package no.nav.emottak.ebms.async.processing
 
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.withContext
 import kotlinx.serialization.encodeToString
 import kotlinx.serialization.json.Json
-import no.nav.emottak.ebms.async.configuration.config
 import no.nav.emottak.ebms.async.log
 import no.nav.emottak.ebms.async.util.EventRegistrationService
 import no.nav.emottak.ebms.validation.CPAValidationService
+import no.nav.emottak.melding.feil.EbmsException
 import no.nav.emottak.message.model.Acknowledgment
-import no.nav.emottak.message.model.EbMSDocument
+import no.nav.emottak.message.model.EbmsMessage
 import no.nav.emottak.message.model.MessageError
-import no.nav.emottak.message.xml.getDocumentBuilder
 import no.nav.emottak.util.marker
 import no.nav.emottak.utils.common.parseOrGenerateUuid
 import no.nav.emottak.utils.kafka.model.EventDataType
 import no.nav.emottak.utils.kafka.model.EventType
-import java.io.ByteArrayInputStream
 
 class SignalMessageService(
     val cpaValidationService: CPAValidationService,
     val eventRegistrationService: EventRegistrationService
 ) {
 
-    suspend fun processSignal(requestId: String, content: ByteArray) {
+    suspend fun processSignal(requestId: String, ebxmlSignalMessage: EbmsMessage) {
         try {
-            val ebxmlSignalMessage = createEbmsMessage(requestId, content)
             eventRegistrationService.registerEventMessageDetails(ebxmlSignalMessage)
-            eventRegistrationService.registerEvent(
-                eventType = EventType.MESSAGE_READ_FROM_QUEUE,
-                requestId = ebxmlSignalMessage.requestId.parseOrGenerateUuid(),
-                messageId = ebxmlSignalMessage.messageId,
-                eventData = Json.encodeToString(
-                    mapOf(EventDataType.QUEUE_NAME.value to config().kafkaSignalReceiver.topic)
-                )
-            )
-            cpaValidationService.validateIncomingMessage(ebxmlSignalMessage)
             when (ebxmlSignalMessage) {
-                is Acknowledgment -> {
-                    processAcknowledgment(ebxmlSignalMessage)
-                }
-                is MessageError -> {
-                    processMessageError(ebxmlSignalMessage)
-                }
+                is Acknowledgment -> processAcknowledgment(ebxmlSignalMessage)
+                is MessageError -> processMessageError(ebxmlSignalMessage)
                 else -> {
                     log.warn(ebxmlSignalMessage.marker(), "Cannot process message as signal message: $requestId")
                     throw RuntimeException("Cannot process message as signal message: $requestId")
```

Comment:
(Trenger ikke å gjøre noe) Tenker høyt: Burde vi lage en egen exception for dette i stedet for å bruke generisk exception? Bare lurer på om det kan være fint å lage beskrivende Exception implemetasjoner, og legge til custom-exceptions nær bruk. Vanligvis er det bare én use case per exception hvis det er veldig spesifikt, unntaksvis fra mer generisk exceptions (NotFound, BadRequest etc)